### ISSUE TITLE
Any plan to add support for miktex?

### ISSUE THREAD

**Issue by lvjr** [teatimeguest/setup-texlive-action#328]
My little tool `texfindpkg` supports both texlive and miktex and I want to run regression tests with GitHub Actions. It would be great if this action could support miktex too. Are there any plans?

**Comment by teatimeguest** [comment]
Thank you for the suggestion! I would like to consider MiKTeX support, but I am lacking knowledge for development because I am not a MiKTeX user. The [official manual](https://docs.miktex.org/manual/) does not seem very helpful for this purpose. Are there any additional resources on MiKTeX internals for developers?

**Comment by pablgonz** [comment]
@teatimeguest @lvjr See https://github.com/koppor/mwe-miktex/blob/main/.github/workflows/miktex.yml (thanks @koppor)

**Comment by lvjr** [comment]
> Just to confirm, this means installing MiKTeX separately from TeX Live, rather than on top of or together with TeX Live, does it not?
****
Yes, and you may decide to make it a separate action "setup-miktex-action".

**Comment by koppor** [comment]
Note that most of the TeX developers recommend using the "TeX Live" distriubtion.

For a full setup, use the docker iamge: https://gitlab.com/islandoftex/images/texlive

For a very small setup, use this action (or [install-texlive](https://github.com/zauguin/install-texlive) by a latex developer. You can use the [DEPP – Dependency Printer for TeX Live](https://gitlab.com/islandoftex/texmf/depp) to determine all required packages.

**Comment by teatimeguest** [comment]
@lvjr I have looked into MiKTeX and found that it has many dependencies, which makes it difficult to install in a portable way. So I think the best way to install MiKTeX is just to use a platform-specific package manager (`apt`, `choco`, `brew`). For an Ubuntu runner, [awalsh128/cache-apt-pkgs-action](https://github.com/awalsh128/cache-apt-pkgs-action) may help.

Nevertheless, I have tried to use `RPATH` on Ubuntu. [This script](https://gist.github.com/teatimeguest/85b85e03153e677e9119a8df12ac6998) installs MiKTeX and its dependencies into a single local directory; X11-related libraries are excluded because they would be broken by `patchelf`, which is not a problem for GitHub-hosted runners since they are pre-installed. But calling `miktexseup` gives the following error:
```console
1 [0x7fb41ae3d0c0] FATAL trace.miktex.core null - The configuration file formats.ini could not be found.
1 [0x7fb41ae3d0c0] FATAL trace.miktex.core null - Data: 
1 [0x7fb41ae3d0c0] FATAL trace.miktex.core null - Source: Libraries/MiKTeX/Core/Session/formats.cpp:163
```
I have not looked into this further yet, but it might be possible to work around and improve it. (Not a recommended method, though)

**Comment by lvjr** [comment]
> I have looked into MiKTeX and found that it has many dependencies, which makes it difficult to install in a portable way. So I think the best way to install MiKTeX is just to use a platform-specific package manager (`apt`, `choco`, `brew`). 

Yes, I agree with you. And it is also the approach in `miktex.yml` mentioned above.

**Comment by koppor** [comment]
> > I have looked into MiKTeX and found that it has many dependencies, which makes it difficult to install in a portable way. So I think the best way to install MiKTeX is just to use a platform-specific package manager (`apt`, `choco`, `brew`).
> Yes, I agree with you. And it is also the approach in `miktex.yml` mentioned above.

Deep link to the respective step: https://github.com/koppor/mwe-miktex/blob/a4444f6362adbd664d800c81357063d12255d904/.github/workflows/miktex.yml#L40. The next lines are required to have MiKTeX's indexes functioning properly.
